modules/affile/affile-source.c

```diff
@@ -253,6 +253,43 @@ affile_sd_construct_proto(AFFileSourceDriver *self, gint fd)
     }
 }
 
+static void
+affile_sd_reset_file_state(PersistState *state, const gchar *old_persist_name)
+{
+  gchar *new_persist_name = g_strdup_printf("%s_DELETED",old_persist_name);
+  persist_state_rename_entry(state,old_persist_name,new_persist_name);
+  g_free(new_persist_name);
+}
+
+static inline gboolean
+_setting_logpipe(LogPipe *s)
+{
+  AFFileSourceDriver *self = (AFFileSourceDriver *) s;
+
+  log_pipe_append((LogPipe *) self->reader, s);
+
+  if (!log_pipe_init((LogPipe *) self->reader))
+    {
+      log_pipe_unref((LogPipe *) self->reader);
+      return FALSE;
+    }
+  return TRUE;
+}
+
+static inline void
+_setting_general_logreader_options(LogPipe *s)
+{
+  AFFileSourceDriver *self = (AFFileSourceDriver *) s;
+
+  log_reader_set_options(self->reader,
+                         s,
+                         &self->reader_options,
+                         STATS_LEVEL1,
+                         SCS_FILE,
+                         self->super.super.id,
+                         self->filename->str);
+}
+
```

Comment:
this could clearly be associated with the _init() operation above, it is actually part of initializing the LogReader, maybe merge that two?